src/shared/ip-protocol-list.c

```diff
@@ -38,3 +40,31 @@ int ip_protocol_from_name(const char *name) {
 int ip_protocol_max(void) {
         return ELEMENTSOF(ip_protocol_names);
 }
+
+int parse_ip_protocol(const char *s) {
+        _cleanup_free_ char *str = NULL;
+        int i, r;
+
+        assert(s);
+
+        if (isempty(s))
+                return IPPROTO_IP;
+
+        /* Do not use strdupa() here */
```

Comment:
why not?

Reply:
Because the input string may come from command line or config files.